lib/debug_logging/class_logger.rb

```diff
@@ -1,72 +1,42 @@
-              method_name: method_to_log,
-              proxy_ref: "kl",
+          define_method(decorated_method) do |*args, **kwargs, &block|
+            lamb_dart = LambDart::Log.new(
+              klass: self,
+              method_config_opts:,
+              method_payload:,
+              args:,
+              kwargs:,
+              decorated_method:,
             )
-            method_return_value = nil
-            log_prefix = nil
-            invocation_id = nil
-            begin
-              config_proxy.log do
-                paydirt = DebugLogging::Util.payload_instance_variable_hydration(scope: self, payload: method_payload)
-                log_prefix = debug_invocation_to_s(
-                  klass: to_s,
-                  separator: "::",
-                  method_to_log: method_to_log,
-                  config_proxy: config_proxy,
-                )
-                invocation_id = debug_invocation_id_to_s(args: args, config_proxy: config_proxy)
-                signature = debug_signature_to_s(args: args, config_proxy: config_proxy)
-                paymud = debug_payload_to_s(payload: paydirt, config_proxy: config_proxy)
-                "#{log_prefix}#{signature}#{invocation_id} debug: #{paymud}"
-              end
-              if config_proxy.benchmarkable_for?(:debug_class_benchmarks)
-                tms = Benchmark.measure do
-                  method_return_value = if args.size == 1 && (harsh = args[0]) && harsh.is_a?(Hash)
-                    original_method.call(**harsh, &block)
-                  else
-                    original_method.call(*args, &block)
-                  end
-                end
-                config_proxy.log do
-                  "#{log_prefix} #{debug_benchmark_to_s(tms: tms)}#{invocation_id}"
-                end
-              else
-                method_return_value = if args.size == 1 && (harsh = args[0]) && harsh.is_a?(Hash)
-                  original_method.call(**harsh, &block)
-                else
-                  original_method.call(*args, &block)
-                end
-                if config_proxy.exit_scope_markable? && invocation_id && !invocation_id.empty?
-                  config_proxy.log do
-                    "#{log_prefix} completed#{invocation_id}"
-                  end
-                end
-              end
-              method_return_value
-            rescue StandardError => e
-              if config_proxy.error_handler_proc
-                config_proxy.error_handler_proc.call(config_proxy, e, self, method_to_log, args)
-              else
-                raise e
+            real_mccoy = ->() {
+              original_method.call(*args, **kwargs, &block)
+            }
+            _dl_ld_enter_log(lamb_dart) do
+              _dl_ld_error_handle(lamb_dart) do
+                return _dl_ld_benchmarked(lamb_dart) { real_mccoy.call } if lamb_dart.bench?
+
+                _dl_ld_exit_log(lamb_dart) { real_mccoy.call }
```

Comment:
@pboling - In this loop over the `methods_to_log`, could we extract this to a separate method (perhaps call it `define_decorated_method`); this way the code is a bit cleaner and potentially faster as it reduces the complexity inside each iteration of the loop.

```ruby
def define_decorated_method(decorated_method, original_method, method_payload, method_config_opts)
    define_method(decorated_method) do |*args, **kwargs, &block|
      lamb_dart = LambDart::Log.new(
        klass: self,
        method_config_opts: method_config_opts,
        method_payload: method_payload,
        args: args,
        kwargs: kwargs,
        decorated_method: decorated_method
      )
      real_mccoy = ->() { original_method.call(*args, **kwargs, &block) }
      _dl_ld_enter_log(lamb_dart) do
        _dl_ld_error_handle(lamb_dart) do
          return _dl_ld_benchmarked(lamb_dart) { real_mccoy.call } if lamb_dart.bench?

          _dl_ld_exit_log(lamb_dart) { real_mccoy.call }
        end
      end
    end
  end
```

And it would be called like this:

```ruby
Array(methods_to_log).each do |decorated_method|

 ...

  (class << self; self; end).class_eval do
      define_decorated_method(decorated_method, original_method, method_payload, method_config_opts)
  end
```

Reply:
Yeah, I've considered doing something like that.  Attacking that complexity is a logical next step.